packages/@aws-cdk/aws-eks/lib/k8s-cluster-autoscaler.ts

```diff
@@ -0,0 +1,389 @@
+              'k8s-app': 'cluster-autoscaler'
+            }
+          },
+          roleRef: {
+            apiGroup: 'rbac.authorization.k8s.io',
+            kind: 'Role',
+            name: 'cluster-autoscaler'
+          },
+          subjects: [
+            {
+              kind: 'ServiceAccount',
+              name: 'cluster-autoscaler',
+              namespace: 'kube-system'
+            }
+          ]
+        },
+        {
+          apiVersion: 'apps/v1',
+          kind: 'Deployment',
+          metadata: {
+            name: 'cluster-autoscaler',
+            namespace: 'kube-system',
+            labels: {
+              app: 'cluster-autoscaler'
+            },
+            annotations: {
+              'cluster-autoscaler.kubernetes.io/safe-to-evict': 'false'
+            }
+          },
+          spec: {
+            replicas: 1,
+            selector: {
+              matchLabels: {
+                app: 'cluster-autoscaler'
+              }
+            },
+            template: {
+              metadata: {
+                labels: {
+                  app: 'cluster-autoscaler'
+                },
+                annotations: {
+                  'prometheus.io/scrape': 'true',
+                  'prometheus.io/port': '8085'
+                }
+              },
+              spec: {
+                serviceAccountName: 'cluster-autoscaler',
+                containers: [
+                   {
+                      image: 'k8s.gcr.io/cluster-autoscaler:' + version,
+                      name: 'cluster-autoscaler',
+                      resources: {
+                         limits: {
+                            cpu: '100m',
+                            memory: '300Mi'
+                         },
+                         requests: {
+                            cpu: '100m',
+                            memory: '300Mi'
+                         }
+                      },
+                      command: [
+                        './cluster-autoscaler',
+                        '--v=4',
+                        '--stderrthreshold=info',
+                        '--cloud-provider=aws',
+                        '--skip-nodes-with-local-storage=false',
+                        '--expander=least-waste',
+                        '--node-group-auto-discovery=asg:tag=k8s.io/cluster-autoscaler/enabled,k8s.io/cluster-autoscaler/' + clusterName,
+                        '--balance-similar-node-groups',
+                        '--skip-nodes-with-system-pods=false'
+                      ],
+                      volumeMounts: [
+                         {
+                            name: 'ssl-certs',
+                            mountPath: '/etc/ssl/certs/ca-certificates.crt',
+                            readOnly: true
+                         }
+                      ],
+                      imagePullPolicy: 'Always'
+                   }
+                ],
+                volumes: [
```

Comment:
Would be nice to provide some references to AWS docs/repos where these resource definitions are coming from, in case someone wants to trace that back in the future.

Generally, it might be nicer to split those up into multiple resources and not one big manifest.

Reply:
Good shout about referencing the docs that this is sourced from. I toyed with splitting it up, but it made readability a bit more difficult. Your earlier comment about using variables for common components I think cleaned it up. Open to ideas as this is the first resource of this kind.

Reply:
See the repo I shard with you... Eventually we could use that